Code/Framework/AzCore/AzCore/std/utility/expected_internal.h

```diff
@@ -0,0 +1,524 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/std/utils.h>
+#include <AzCore/std/typetraits/is_assignable.h>
+#include <AzCore/std/typetraits/is_constructible.h>
+#include <AzCore/std/typetraits/is_destructible.h>
+#include <AzCore/std/typetraits/is_void.h>
+
+namespace AZStd
+{
+    template <class T, class E>
+    class expected;
+
+    template<class E>
+    class unexpected;
+
+    //! Tag Type for in-place construction of the error type
+    struct unexpect_t { explicit unexpect_t() = default; };
+    inline constexpr unexpect_t unexpect{};
+}
+
+namespace AZStd::Internal
+{
+    template<class T, class E, class U, class G>
+    constexpr bool not_convertible_or_constructible_from_other_std_expected_v
+        = !is_constructible_v<T, expected<U, G>&>
+        || !is_constructible_v<T, expected<U, G>>
+        || !is_constructible_v<T, const expected<U, G>&>
+        || !is_constructible_v<T, const expected<U, G>>
+        || !is_convertible_v<expected<U, G>&, T>
+        || !is_convertible_v<expected<U, G>&&, T>
+        || !is_convertible_v<const expected<U, G>&, T>
+        || !is_convertible_v<const expected<U, G>&&, T>
+        || !is_constructible_v<unexpected<E>, expected<U, G>&>
+        || !is_constructible_v<unexpected<E>, expected<U, G>>
+        || !is_constructible_v<unexpected<E>, const expected<U, G>&>
+        || !is_constructible_v<unexpected<E>, const expected<U, G>>;
+
+    template<class T>
+    constexpr bool is_std_unexpected_specialization_v = false;
+
+    template<class E>
+    constexpr bool is_std_unexpected_specialization_v<unexpected<E>> = true;
+
+    // Due to using not having C++20 support, the requires block cannot be used
+    // to specify constraints on a class special member functions(default constructor, copy/move constructor, destructor)
+    // without using function templates on the AZStd::expected class template.
+    // The problem with using a function template though, is that it isn't default'able.
```

Comment:
nit: default'able -> defaultable?

Reply:
Or maybe this is the accepted way of writing it as I see it appearing elsewhere

Reply:
Well I don't think `defaultable` is a word, so I put the apostrophe to separate it